### PR TITLE
Interpolate displayVersion into the right place in the tree

### PR DESCRIPTION
**Pull request by blast-hardcheese**
Why
===

Top-level `displayVersion` is actually the semantically-relevant minified version, `.runner.[*].displayVersion` is the full version string

What changed
============

Interpolate values from the new `override` parameter

Test plan
=========

```
$ nix build .#'"nodejs-16"'
$ jq . < result | jq '.displayVersion, .runners.nodeJS.displayVersion'
"16"
"16.18.1"
```

Rollout
=======

_Describe any procedures or requirements needed to roll this out safely (or check the box below)_

- [x] This is fully backward and forward compatible

### CODE REVIEW

File: pkgs/historical-modules/default.nix [hunk 1]
```diff
@@ -11,106 +11,193 @@ let
...
+      };
     }
     {
       moduleId = "go-1.19";
       commit = "c48c43c6c698223ed3ce2abc5a2d708735a77d5b";
-      displayVersion = "1.19.3";
+      overrides = {
+        # /nix/store/7icmam9gmbfas0zbpm53f3ilw79kx0rj-replit-module-go-1.19
+        # .runners["go-run"].displayVersion = "1.19.3";
+        displayVersion = "1";
```
Comment (ryantm) [inline pkgs/historical-modules/default.nix:69]: For go, I believe the minor version matters quite a bit.

File: pkgs/historical-modules/default.nix [hunk 2]
```diff
@@ -11,106 +11,193 @@ let
...
+      overrides = {
+        # /nix/store/65v1j76svgdi8bj3sxx5hfshbvdyissl-replit-module-zig-0.11
+        # .runners["zig-build-run"].displayVersion = "0.11.0";
+        displayVersion = "0";
+      };
     }
   ];
 
-  moduleFromHistory = { displayVersion, moduleId, commit, deployment ? false }:
+  moduleFromHistory = { moduleId, commit, deployment ? false, overrides }:
```
Comment (ryantm) [inline pkgs/historical-modules/default.nix:200]: We need to fix line 143 below too

File: pkgs/historical-modules/default.nix [hunk 3]
```diff
@@ -11,106 +11,193 @@ let
...
+        displayVersion = "5";
+      };
     }
     {
       moduleId = "vue-node-18";
       commit = "3ea4bcdbdc3c5e3c09b37b07edcd61781f9695f7";
-      displayVersion = "18.18.2";
+      overrides = {
+        # /nix/store/623sn0cy54r5csab3d9fmg55di6r4dd3-replit-module-vue-node-18
+        # .runners["dev-runner"].displayVersion = "18.18.2";
```
Comment (airportyh) [inline pkgs/historical-modules/default.nix:185]: Why are some of these commented out?

Comment (blast-hardcheese) [reply pkgs/historical-modules/default.nix:185]: I spot-checked and it didn't seem as though they were used, I figured I'd just fix the ones that I actually need right now and we can add the rest of the specific versions later if it turns out they're useful. I did the legwork of correlating the values and keys with the comments, so it should be easier in the next pass.

**Comment by blast-hardcheese** [comment]
I believe this is now good to go.

File: pkgs/historical-modules/default.nix (hunk 3 above)
Comment (airportyh) [reply pkgs/historical-modules/default.nix:185]: ok

**Review by airportyh — APPROVED** [review]

**Review by ryantm — APPROVED** [review]

**Pull request merged**